CorpusCallosum/fastsurfer_cc.py

```diff
@@ -0,0 +1,880 @@
+    fsaverage_header : nibabel.Nifti1Header
+        FSAverage header fields for LTA writing.
+    vox2ras_tkr : np.ndarray
+        Voxel to RAS tkr-space transformation matrix.
+
+    Notes
+    -----
+    The function uses pre-computed fsaverage centroids and data from static files
+    to perform the registration. It matches corresponding anatomical structures
+    between the subject's segmentation and fsaverage space.
+    """
+    if verbose:
+        print("Centroid registration")
+
+    # Load pre-computed fsaverage centroids and data from static files
+    centroids_dst = load_fsaverage_centroids(FSAVERAGE_CENTROIDS_PATH)
+    fsaverage_affine, fsaverage_header, vox2ras_tkr = load_fsaverage_data(FSAVERAGE_DATA_PATH)
+
+    centroids_mov, ids_not_found = get_centroids_from_nib(aseg_nib, label_ids=list(centroids_dst.keys()))
+
+    # delete not found labels from centroids_mov
+    for id in ids_not_found:
+        del centroids_dst[id]
+
+    centroids_mov = np.array(list(centroids_mov.values())).T
+    centroids_dst = np.array(list(centroids_dst.values())).T
+
+    orig_fsaverage_ras2ras = find_rigid(p_mov=centroids_mov.T, p_dst=centroids_dst.T)
+
+    # make affine that increases resolution to orig resolution
+    resolution_orig = aseg_nib.header.get_zooms()[0]
+    resolution_trans = np.eye(4)
+    resolution_trans[0, 0] = resolution_orig
+    resolution_trans[1, 1] = resolution_orig
+    resolution_trans[2, 2] = resolution_orig
+
+    orig_fsaverage_vox2vox = (
+        np.linalg.inv(resolution_trans @ fsaverage_affine) @ orig_fsaverage_ras2ras @ aseg_nib.affine
+    )
+    fsaverage_hires_affine = resolution_trans @ fsaverage_affine
+
+    return orig_fsaverage_vox2vox, orig_fsaverage_ras2ras, fsaverage_hires_affine, fsaverage_header, vox2ras_tkr
+
+
+def localize_ac_pc(
+    midslices: np.ndarray,
+    aseg_nib: "nib.Nifti1Image",
+    orig_fsaverage_vox2vox: np.ndarray,
+    model_localization: "torch.nn.Module",
+    slices_to_analyze: int
+) -> tuple[np.ndarray, np.ndarray]:
+    """Localize anterior and posterior commissure points in the brain.
+
+    Uses a trained model to detect AC and PC points in mid-sagittal slices,
+    using the third ventricle as an anatomical reference.
+
+    Parameters
+    ----------
+    midslices : np.ndarray
+        Array of mid-sagittal slices.
+    aseg_nib : nibabel.Nifti1Image
+        Subject's segmentation image.
+    orig_fsaverage_vox2vox : np.ndarray
+        Transformation matrix to fsaverage space.
+    model_localization : torch.nn.Module
+        Trained model for AC-PC detection.
+    slices_to_analyze : int
+        Number of slices to process.
+
+    Returns
+    -------
+    ac_coords : np.ndarray
+        Coordinates of the anterior commissure.
+    pc_coords : np.ndarray
+        Coordinates of the posterior commissure.
+    """
+
+    # get center of third ventricle from aseg and map to fsaverage space
+    third_ventricle_mask = aseg_nib.get_fdata() == 4
```

Comment:
```suggestion
    third_ventricle_mask = np.asarray(aseg_nib.dataobj) == 4
```
Ideally, the third ventricle index would get read from the ColorLUT